docs/source/en/quicktour.mdx

```diff
@@ -118,20 +118,27 @@ Create a [`pipeline`] with the task you want to solve for and the model you want
 Next, load a dataset (see the 🤗 Datasets [Quick Start](https://huggingface.co/docs/datasets/quickstart.html) for more details) you'd like to iterate over. For example, let's load the [MInDS-14](https://huggingface.co/datasets/PolyAI/minds14) dataset:
 
 ```py
->>> from datasets import load_dataset
+>>> from datasets import load_dataset, Audio
 
 >>> dataset = load_dataset("PolyAI/minds14", name="en-US", split="train")  # doctest: +IGNORE_RESULT
 ```
 
+We need to make sure that the sampling rate of the dataset matches the sampling 
+rate `facebook/wav2vec2-base-960h` was trained on.
+
+```py
+>>> dataset = dataset.cast_column("audio", Audio(sampling_rate=speech_recognizer.feature_extractor.sampling_rate))
+```
+
 You can pass a whole dataset pipeline:
 
 ```py
->>> files = dataset["path"]
->>> speech_recognizer(files[:4])
-[{'text': 'I WOULD LIKE TO SET UP A JOINT ACCOUNT WITH MY PARTNER HOW DO I PROCEED WITH DOING THAT'}, 
- {'text': "FONDERING HOW I'D SET UP A JOIN TO HELL T WITH MY WIFE AND WHERE THE AP MIGHT BE"}, 
- {'text': "I I'D LIKE TOY SET UP A JOINT ACCOUNT WITH MY PARTNER I'M NOT SEEING THE OPTION TO DO IT ON THE APSO I CALLED IN TO GET SOME HELP CAN I JUST DO IT OVER THE PHONE WITH YOU AND GIVE YOU THE INFORMATION OR SHOULD I DO IT IN THE AP AN I'M MISSING SOMETHING UQUETTE HAD PREFERRED TO JUST DO IT OVER THE PHONE OF POSSIBLE THINGS"}, 
- {'text': 'HOW DO I FURN A JOINA COUT'}]
+>>> raw_audio_waveforms = [d["array"] for d in dataset[:4]["audio"]]
```

Comment:
Can we select the four first indices then still pass a dataset? Using a list like this contradicts the text above (you can pass a whole dataset to a pileline).

Reply:
I think the blocker for this is that the pipeline expects

- A raw `ndarray` (here `d["array"]`)
- A dict of `{"raw": ndarray, "sampling_rate": 16_000}` but `datasets` creates `{"array": ndarray, "sampling_rate": 16_000}`

```python
speech_recognizer(dataset["audio"]["array"])
....
```
Should work no ? (since the sampling rate is already taken care of beforehand)

Reply:
`dataset["audio"]` loads + resamples the whole dataset. For audio it's important to first slice the list and then call the columns.

We could use `dataset.select(...)` to remedy this but it'd still not work, e.g.:

```py
from datasets import load_dataset

ds = load_dataset("PolyAI/minds14", "en-US", split="train")
ds = ds.select(range(4))

ds["audio"]["array"] # <- gives TypeError: list indices must be integers or slices, not str
```

Think the best I can do here is to rewrite the text a bit, no?

cc @lhoestq @albertovilla @polinaeterna @anton-l here as well - think that's a common problem we have with `ds["audio"]["array"]`

Reply:
Ok for me to rewrite the text and say it can take list/array/dataset. But the previous example looked simpler :-p

Reply:
You can't do `ds[:4]["audio"]["array"]`. Indeed in the general case, the sampling rate may not be the same for all samples: `sampling_rate` is an optional parameter of `Audio`. So the format of `ds[:4]["audio"]` is a list of dicts {"array":..., "sampling_rate":...}

Reply:
We can update the `pipeline` to accept both `raw` and `array`. Seems like the path of least resistance here, wdyt ?

Reply:
Changing the text and merging for now though